nri: update NRI to v0.10.0.

klihub opened the pull request:
This patch set
- bumps NRI to the latest tagged v0.10.0 version
- adds configuration for the built-in NRI default validator plugin
- warns (in the current implementation repeatedly) if old v0.1.0-style NRI plugins are found
- fixes test util script to generate NRI configuration with valid configuration fields only

chrishenzie commented on internal/cri/server/podsandbox/opts.go:
```diff
@@ -32,20 +33,35 @@ func WithNRISandboxDelete(sandboxID string) containerd.ProcessDeleteOpts {
+	}
+
+	if plugins := nric.Plugins(); len(plugins) != 0 {
+		log.G(ctx).Warnf("Deprecated NRI 0.1.0-style plugins configured: %s", strings.Join(plugins, ","))
```
I opened a PR for this previously here, in case we can draw inspiration from it:
https://github.com/containerd/containerd/pull/11951

My idea here was to emit a deprecation warning like we did for the other deprecated features.

klihub commented:
/test pull-containerd-k8s-e2e-ec2

chrishenzie reviewed (approved):
Other than one nit LGTM

klihub commented on internal/cri/server/podsandbox/sandbox_run.go:
```diff
@@ -261,16 +263,21 @@ func (c *Controller) Start(ctx context.Context, id string) (cin sandbox.Controll
-		nriSB := &nri.Sandbox{
+		if plugins := nric.Plugins(); len(plugins) != 0 { //nolint:staticcheck
+			c.warningService.Emit(ctx, deprecation.NRIV010Plugin)
+			log.G(ctx).Warnf("Deprecated NRI v0.1.0-style plugins: %s", strings.Join(plugins, ","))
```
Okay, I'll add more clarification there.

klihub replied:
> ... (in case they don't invoke the deprecation service to get full details)

@chrishenzie Then again, I'd just add another log message here with a note about having to (soon) use the v0.1.0 adapter. But that's already mentioned in the emitted deprecation warning on the previous line, so I doubt there would be any tangible benefit of repeating it again. WDYT ?

klihub replied:
@chrishenzie I updated to reuse the verbatim deprecation message in the logged warning. Now it looks something like
```
Aug 21 18:35:18 n4c16-fedora-40-cloud-base-containerd containerd[58502]: time="2025-08-21T18:35:18.755785167Z" level=warning msg="Deprecated NRI plugin(s) /opt/nri/bin/test: NRI 0.1.0-style plugins are deprecated since containerd 2.2 and should only be used through the v010-adapter plugin."
```

mikebrow reviewed (commented):
could we get an update on https://github.com/containerd/containerd/blob/main/docs/NRI.md e.g. for the validator configuration 

On the deprecation the NRIv010 plugins.. we never officially supported NRI as stable service api rather it is in the experimental list, but I suppose it would be ok to list the deprecation in the table https://github.com/containerd/containerd/blob/main/RELEASES.md#deprecated-features

klihub commented:
> could we get an update on https://github.com/containerd/containerd/blob/main/docs/NRI.md e.g. for the validator configuration
> 
> On the deprecation the NRIv010 plugins.. we never officially supported NRI as stable service api rather it is in the experimental list, but I suppose it would be ok to list the deprecation in the table https://github.com/containerd/containerd/blob/main/RELEASES.md#deprecated-features

Added v0.1.0 NRI plugin support to the table of deprecated features. Added a chapter about how to enable and configure the default validator to the docs.

@mikebrow We need to think how to best handle documentation for NRI. I think probably the best would be to have any detailed documentation only present in the NRI repo itself, and only keep a short overview, pointers to the NRI documentation, and description of the runtime-specific NRI configuration bits on the runtime side, so we can minimize duplicated (and eventually stale) information across NRI, containerd and CRI-O.

mikebrow commented:
> > could we get an update on https://github.com/containerd/containerd/blob/main/docs/NRI.md e.g. for the validator configuration
> > On the deprecation the NRIv010 plugins.. we never officially supported NRI as stable service api rather it is in the experimental list, but I suppose it would be ok to list the deprecation in the table https://github.com/containerd/containerd/blob/main/RELEASES.md#deprecated-features
> 
> Added v0.1.0 NRI plugin support to the table of deprecated features. Added a chapter about how to enable and configure the default validator to the docs.
> 
> @mikebrow We need to think how to best handle documentation for NRI. I think probably the best would be to have any detailed documentation only present in the NRI repo itself, and only keep a short overview, pointers to the NRI documentation, and description of the runtime-specific NRI configuration bits on the runtime side, so we can minimize duplicated (and eventually stale) information across NRI, containerd and CRI-O.

nod to minimal summary in containerd/docs/NRI.md with pointers being more than sufficient

mikebrow reviewed (approved):
LGTM
